features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64LRUCache.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import android.util.LruCache
+import androidx.annotation.VisibleForTesting
+
+internal object Base64LRUCache : Cache {
+    private const val MAX_ITEMS = 1000
+    private var cache = LruCache<Int, String>(MAX_ITEMS)
```

Comment:
are you sure you want to base it on number of items ? Why not on bytes ?

Reply:
You could leverage the :
`Runtime.maxMemory` which returns the max amount of memory in bytes allocated for current JVM and then take a part of it (e.g. Runtime.maxMemory/8)